Zend/zend_API.c

```diff
@@ -4097,29 +4097,41 @@ ZEND_API int zend_update_static_property_stringl(zend_class_entry *scope, const
 }
 /* }}} */
 
-ZEND_API zval *zend_read_property_ex(zend_class_entry *scope, zval *object, zend_string *name, zend_bool silent, zval *rv) /* {{{ */
-{
+
+ZEND_API zval *zend_obj_read_property_ex(zend_class_entry *scope, zend_object *object, zend_string *name, zend_bool silent, zval *rv)
```

Comment:
Please commit these new APIs separately.

Reply:
Why do we need this duplication?

Reply:
The existing API don't actually need `zval`, and in some places (notably most places dealing with exceptions) it forces you to make a zval just to pass it to a function that doesn't actually need a `zval`.